<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateRobotConfigDialog</class>
 <widget class="QDialog" name="CreateRobotConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>728</width>
    <height>567</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>540</y>
     <width>156</width>
     <height>23</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>9</y>
     <width>711</width>
     <height>77</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetFixedSize</enum>
    </property>
    <item row="1" column="2">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>图片</string>
      </property>
     </widget>
    </item>
    <item row="1" column="5">
     <widget class="QPushButton" name="setpng">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>设置图片</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>厂家</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="author"/>
    </item>
    <item row="0" column="3" colspan="3">
     <widget class="QLineEdit" name="factory"/>
    </item>
    <item row="2" column="2">
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>关节数</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="description"/>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="name"/>
    </item>
    <item row="1" column="4">
     <widget class="QPushButton" name="delpng">
      <property name="text">
       <string>删除图片</string>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QLineEdit" name="joint_number"/>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>描述</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>名字</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>作者</string>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QLineEdit" name="png">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="2" column="4">
     <widget class="QCheckBox" name="externAxis_checkBox">
      <property name="text">
       <string>使用扩展轴</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="edit_enable">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>540</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>编辑</string>
   </property>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>90</y>
     <width>711</width>
     <height>451</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>2</number>
   </property>
   <widget class="QWidget" name="model_tab">
    <attribute name="title">
     <string>模型文件</string>
    </attribute>
    <widget class="QListView" name="stepfiles">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>691</width>
       <height>381</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>224</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>模型文件：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="add">
        <property name="text">
         <string>选择</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="remove">
        <property name="text">
         <string>清空</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="joint_tab">
    <attribute name="title">
     <string>关节参数</string>
    </attribute>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>701</width>
       <height>391</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>699</width>
        <height>389</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QCheckBox" name="dh_enable_checkBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>使用自建模型</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="dh_param_checkbox">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>10</y>
       <width>141</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>使用DH参数</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="lua_tab">
    <attribute name="title">
     <string>Lua脚本</string>
    </attribute>
    <widget class="QLabel" name="test_lua_result">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>621</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QWidget" name="">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>691</width>
       <height>421</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>测试数据</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="test_lua_data"/>
        </item>
        <item>
         <widget class="QPushButton" name="test_lua">
          <property name="text">
           <string>测试Lua</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>测试结果</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCodeEditor" name="luatextEdit"/>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="tab">
    <attribute name="title">
     <string>默认lua脚本</string>
    </attribute>
    <widget class="QCodeEditor" name="plainTextEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>681</width>
       <height>411</height>
      </rect>
     </property>
     <property name="plainText">
      <string>-- 请在内部完善函数体，不要修改函数名和参数，需要返回一个浮点数组

-- 更新关节坐标
function hc_updateJoints(joints_in) 
-- 请在此处写 
-- 定义一个数组,输出关节坐标 
local joints_out = {} 
 
-- 给数组赋值 
joints_out[1] = joints_in[1] 
joints_out[2] = joints_in[2] 
joints_out[3] = joints_in[3] 
joints_out[4] = joints_in[4] 
joints_out[5] = joints_in[5] 
joints_out[6] = joints_in[6] 
 
return joints_out 
end

-- 更新基座坐标
function hc_updateBaseJoints(joints_in) 
-- 请在此处写 
-- 定义一个数组,输出基座坐标 长度为6
local joints_out = {} 
 
-- 给数组赋值 
joints_out[1] = joints_in[7] 
joints_out[2] = 0 
joints_out[3] = 0 
joints_out[4] = 0 
joints_out[5] = 0 
joints_out[6] = 0 
 
return joints_out 
end</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">QCodeEditor.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CreateRobotConfigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CreateRobotConfigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
